File: C#/Microsoft Office/PSSDK9-CSharp/Form1.cs
using System;
using System.IO;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using pswift = PlanSwift9;
using Word = Microsoft.Office.Interop.Word;
using Excel = Microsoft.Office.Interop.Excel;
using Outlook = Microsoft.Office.Interop.Outlook;

namespace WindowsFormsApplication1
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
        public void itmtoword(pswift.IItem itm, Word.Table tbl, int rowidx, string tp)
        {
            if (rowidx > 20)
            {
                tbl.Rows.Add(System.Reflection.Missing.Value);
            }
            if (tp == "Digitizer")
            {
                tbl.Cell(rowidx, 1).Range.Font.Bold = 1;
                tbl.Cell(rowidx, 1).Range.Font.Italic = 0;
            }
            else
            {
                tbl.Cell(rowidx, 1).Range.Font.Bold = 0;
                tbl.Cell(rowidx, 1).Range.Font.Italic = 1;
            }
            tbl.Cell(rowidx, 1).Range.Text = itm.Name;
            tbl.Cell(rowidx, 2).Range.Text = itm.GetPropertyResultAsString("Qty","");
            tbl.Cell(rowidx, 3).Range.Text = itm.GetProperty("Qty").Units;
            tbl.Cell(rowidx, 4).Range.Text = itm.GetPropertyResultAsString("Price Each","");
            tbl.Cell(rowidx, 5).Range.Text = itm.GetPropertyResultAsString("Price Total","");
        }
        private void button1_Click(object sender, EventArgs e)
        {
            Form2 RForm = new Form2();
            RForm.ShowDialog();
            if (RForm.DialogResult != DialogResult.OK){
                RForm.Dispose();
     
                return;
            }
            Object template = Directory.GetCurrentDirectory() + "\\Includes\\Word_Template.dotx";
            Console.WriteLine(template.ToString());
            Object oMissing = System.Reflection.Missing.Value;
            Object isvisible = true;
            Word._Application oWord = new Word.Application();
            try
            {
                Word._Document oDoc = oWord.Documents.Add(template, oMissing, oMissing, isvisible);
                Word.Table otable = oDoc.Tables[2];
                String ReportType = RForm.ReportCBX.Text;
                if (ReportType == "Digitizer Items Only")
                {
                    int rowidx = 1;
                    for (int idx = 0; idx <= TGlobal.tlst.Count - 1; idx++)
                    {
                        pswift.IItem itm = TGlobal.ps.GetItem(TGlobal.tlst[idx]);
                        Boolean isDigitizer = false;
                        if (itm.GetPropertyResultAsBoolean("isArea", false) == true)
                        {
                            isDigitizer = true;
                        }
                        if (itm.GetPropertyResultAsBoolean("isLinear", false) == true)
                        {
                            isDigitizer = true;
                        }
                        if (itm.GetPropertyResultAsBoolean("isSegment", false) == true)
                        {
                            isDigitizer = true;
                        }
                        if (itm.GetPropertyResultAsBoolean("isCount", false) == true)
                        {
                            isDigitizer = true;
                        }
                        
                        if (isDigitizer) {
                            rowidx++;
                            itmtoword(itm, otable, rowidx, "Digitizer");


                        }
                        this.progressBar1.Value = idx;      
                    }
                }
                if (ReportType == "Parts Only")
                {
                    int rowidx = 1;
                    for (int idx = 0; idx <= TGlobal.tlst.Count - 1; idx++)
                    {
                        pswift.IItem itm = TGlobal.ps.GetItem(TGlobal.tlst[idx]);
                        if (itm.GetPropertyResultAsBoolean("isPart", false) == true) 
                        {
                            rowidx++;
                             itmtoword(itm,otable,rowidx,"Part");
                        }
                        this.progressBar1.Value = idx;
                    }
                }
                if (ReportType == "Digitizer Items w/Parts")
                {
                    int rowidx = 1;
                    for (int idx = 0; idx <= TGlobal.tlst.Count - 1; idx++)
                    {
                        pswift.IItem itm = TGlobal.ps.GetItem(TGlobal.tlst[idx]);
                      
                            rowidx++;
                            if (itm.GetPropertyResultAsBoolean("isPart", false) == true)
                            {
                                itmtoword(itm, otable, rowidx, "Part");
                            }
                            else
                            {
                                itmtoword(itm, otable, rowidx, "Digitizer");
                            }

                            this.progressBar1.Value = idx;
                    }
                }
                
            }
            finally
            {
                this.progressBar1.Value = 0;
                oWord.Visible = true;
                oWord = null;
                RForm.Dispose();

            }


        }

        public void loadTakeoffItems(pswift.IItem itm, List<string> lst)
        {
            for (int idx = 0; idx <= itm.ChildCount() - 1; idx++)
            {
                pswift.IItem citm = itm[idx];
                if (citm.GetProperty("Type").ResultAsString() == "Folder")
                {
                    loadTakeoffItems(citm,lst);
                }
                else
                {
                    Boolean isItem = citm.GetPropertyResultAsBoolean("IsItem", false);
                    if (isItem)
                    {
                        lst.Add(citm.GUID());
                    }
                    if (citm.ChildCount() != 0)
                    {
                        loadTakeoffItems(citm, lst);
                    }
                    
                }

            } 
        }
        private void Form1_Load(object sender, EventArgs e)
        {
            TGlobal.ps = new pswift.PlanSwift();
            string tkpath = TGlobal.ps.Root().FullPath() + "\\Job\\TakeOff";
            pswift.IItem takeoff = TGlobal.ps.GetItem(tkpath);
            TGlobal.tlst = new List<string>();
           
            loadTakeoffItems(takeoff,TGlobal.tlst);
            this.progressBar1.Minimum = 0;
            this.progressBar1.Maximum = TGlobal.tlst.Count - 1;
            
        }
      

        class TGlobal
        {
            public static pswift.IPlanSwift ps;
            public static List<string> tlst;
        }
        public void itmtoexcel(pswift.IItem itm, int rowidx, Excel._Worksheet osheet)
        {
            if (rowidx > 36)
            {
                osheet.Cells[rowidx, 1].EntireRow.Insert(System.Reflection.Missing.Value);
            }
            osheet.Cells[rowidx, 1].value = itm.GetPropertyResultAsString("Qty", "");
            osheet.Cells[rowidx, 2].value = itm.GetProperty("Qty").Units;
            osheet.Cells[rowidx, 3].value = itm.GetPropertyResultAsString("Item #","");
            osheet.Cells[rowidx, 4].value = itm.Name;
            osheet.Cells[rowidx,5].value = itm.GetPropertyResultAsString("Price Each","");
            osheet.Cells[rowidx, 6].value = itm.GetPropertyResultAsString("Price Total", "");
        }

        private void button2_Click(object sender, EventArgs e)
        {
            Form2 RForm = new Form2();
            RForm.ShowDialog();
            if (RForm.DialogResult != DialogResult.OK)
            {
                RForm.Dispose();

                return;
            }
            Object template = Directory.GetCurrentDirectory() + "\\Includes\\Excel_Template.XLT";
            Console.WriteLine(template.ToString());
            Object oMissing = System.Reflection.Missing.Value;
            Object isvisible = true;
            Excel._Application oExcel = new Excel.Application();
            try
            {
                Excel._Workbook oBook = oExcel.Workbooks.Add(template);
                Excel._Worksheet oSheet = oExcel.Worksheets[1];
                String ReportType = RForm.ReportCBX.Text;
                if (ReportType == "Digitizer Items Only")
                {
                    int rowidx = 17;
                    for (int idx = 0; idx <= TGlobal.tlst.Count - 1; idx++)
                    {
                        pswift.IItem itm = TGlobal.ps.GetItem(TGlobal.tlst[idx]);
                        Boolean isDigitizer = false;
                        if (itm.GetPropertyResultAsBoolean("isArea", false) == true)
                        {
                            isDigitizer = true;
                        }
                        if (itm.GetPropertyResultAsBoolean("isLinear", false) == true)
                        {
                            isDigitizer = true;
                        }
                        if (itm.GetPropertyResultAsBoolean("isSegment", false) == true)
                        {
                            isDigitizer = true;
                        }
                        if (itm.GetPropertyResultAsBoolean("isCount", false) == true)
                        {
                            isDigitizer = true;
                        }

                        if (isDigitizer)
                        {
                            rowidx++;
                            itmtoexcel(itm, rowidx, oSheet);


                        }
                        this.progressBar1.Value = idx;
                    }
                }
                if (ReportType == "Parts Only")
                {
                    int rowidx = 17;
                    for (int idx = 0; idx <= TGlobal.tlst.Count - 1; idx++)
                    {
                        pswift.IItem itm = TGlobal.ps.GetItem(TGlobal.tlst[idx]);
                        if (itm.GetPropertyResultAsBoolean("isPart", false) == true)
                        {
                            rowidx++;
                            itmtoexcel(itm, rowidx,oSheet);
                        }
                        this.progressBar1.Value = idx;
                    }
                }
                if (ReportType == "Digitizer Items w/Parts")
                {
                    int rowidx = 17;
                    for (int idx = 0; idx <= TGlobal.tlst.Count - 1; idx++)
                    {
                        pswift.IItem itm = TGlobal.ps.GetItem(TGlobal.tlst[idx]);

                        rowidx++;
                        if (itm.GetPropertyResultAsBoolean("isPart", false) == true)
                        {
                            itmtoexcel(itm, rowidx, oSheet);
                        }
                        else
                        {
                            itmtoexcel(itm, rowidx, oSheet);
                        }

                        this.progressBar1.Value = idx;
                    }
                }

            }
            finally
            {
                this.progressBar1.Value = 0;
                oExcel.Visible = true;
                oExcel = null;
                RForm.Dispose();

            }
        }
        public string createMailBody(string RT)
        {
            char quo = (char)34;
             string body = "<body>";
                body = body + "<h1>Report Type: " + RT + "</h1>" + "\r\n";
                body = body + "<table style="+ quo +"width:auto; font-weight:Bold; font-family:Tahoma; font-size:14px;"+ quo +">" + "\r\n";
                body = body + "<tr>" + "\r\n";
                body = body + "<td style="+ quo +"width:200px; font-weight:Bold; font-family:Tahoma; font-size:14px;"+ quo +">";
                body = body + "Name";
                body = body + "</td>" + "\r\n";
                body = body + "<td style="+ quo +"width:100px; font-weight:Bold; font-family:Tahoma; font-size:14px;"+ quo +">";
                body = body + "Qty";
                body = body + "</td>" + "\r\n";
                body = body + "<td style="+ quo +"width:100px; font-weight:Bold; font-family:Tahoma; font-size:14px;"+ quo +">";
                body = body + "Price Each";
                body = body + "</td>" + "\r\n";
                body = body + "<td style="+ quo +"width:100px; font-weight:Bold; font-family:Tahoma;font-size:14px;"+ quo +">";
                body = body + "Price Total";
                body = body + "</td>" + "\r\n";
                body = body + "</tr>" + "\r\n";
                if (RT == "Digitizer Items Only")
                {
                for (int idx = 0; idx <=  TGlobal.tlst.Count -1; idx++){
                    pswift.IItem itm = TGlobal.ps.GetItem(TGlobal.tlst[idx]);
                    if (itm.GetPropertyResultAsBoolean("isPart",false) == false) {
                        body = body + "<tr><td style="+ quo +"font-size:12px; font-weight:normal;"+ quo +">" + itm.Name + "</td>";
                        body = body + "<td style="+ quo +"font-size:12px; font-weight:normal;"+ quo +">" + itm.GetPropertyResultAsString("Qty","") + "</td>";
                        body = body + "<td style="+ quo +"font-size:12px; font-weight:normal;"+ quo +">" + itm.GetPropertyResultAsString("Price Each","") + "</td>";
                        body = body + "<td style=" + quo + "font-size:12px; font-weight:normal;" + quo + ">" + itm.GetPropertyResultAsString("Price Total","") + "</td>";
                        body = body + "</tr>";
                    }

                }
                

                }
                if (RT == "Parts Only")
                {
                    for (int idx = 0; idx <= TGlobal.tlst.Count - 1; idx++)
                    {
                        pswift.IItem itm = TGlobal.ps.GetItem(TGlobal.tlst[idx]);
                        if (itm.GetPropertyResultAsBoolean("isPart", false) == true)
                        {
                            body = body + "<tr><td style=" + quo + "font-size:12px; font-weight:normal;" + quo + ">" + itm.Name + "</td>";
                            body = body + "<td style=" + quo + "font-size:12px; font-weight:normal;" + quo + ">" + itm.GetPropertyResultAsString("Qty", "") + "</td>";
                            body = body + "<td style=" + quo + "font-size:12px; font-weight:normal;" + quo + ">" + itm.GetPropertyResultAsString("Price Each", "") + "</td>";
                            body = body + "<td style=" + quo + "font-size:12px; font-weight:normal;" + quo + ">" + itm.GetPropertyResultAsString("Price Total", "") + "</td>";
                            body = body + "</tr>";
                        }

                    }


                }
                if (RT == "Digitizer Items w/Parts")
                {
                    for (int idx = 0; idx <= TGlobal.tlst.Count - 1; idx++)
                    {
                        pswift.IItem itm = TGlobal.ps.GetItem(TGlobal.tlst[idx]);
                        if (itm.GetPropertyResultAsBoolean("isPart", false) == false)
                        {
                            body = body + "<tr><td style=" + quo + "font-size:12px; font-weight:normal; color:#007dc3;" + quo + ">" + itm.Name + "</td>";
                            body = body + "<td style=" + quo + "font-size:12px; font-weight:normal; color:#007dc3;" + quo + ">" + itm.GetPropertyResultAsString("Qty", "") + "</td>";
                            body = body + "<td style=" + quo + "font-size:12px; font-weight:normal; color:#007dc3;" + quo + ">" + itm.GetPropertyResultAsString("Price Each", "") + "</td>";
                            body = body + "<td style=" + quo + "font-size:12px; font-weight:normal; color:#007dc3;" + quo + ">" + itm.GetPropertyResultAsString("Price Total", "") + "</td>";
                            body = body + "</tr>";
                        }
                        if (itm.GetPropertyResultAsBoolean("isPart", false) == true)
                        {
                            body = body + "<tr><td style=" + quo + "font-size:12px; font-weight:normal; color:#0000FF;" + quo + ">" + itm.Name + "</td>";
                            body = body + "<td style=" + quo + "font-size:12px; font-weight:normal; color:#0000FF;" + quo + ">" + itm.GetPropertyResultAsString("Qty", "") + "</td>";
                            body = body + "<td style=" + quo + "font-size:12px; font-weight:normal; color:#0000FF;" + quo + ">" + itm.GetPropertyResultAsString("Price Each", "") + "</td>";
                            body = body + "<td style=" + quo + "font-size:12px; font-weight:normal; color:#0000FF;" + quo + ">" + itm.GetPropertyResultAsString("Price Total", "") + "</td>";
                            body = body + "</tr>";
                        }

                    }


                }
            return body;
        }
        private void button3_Click(object sender, EventArgs e)
        {
            Form2 RForm = new Form2();
            RForm.ShowDialog();
            if (RForm.DialogResult != DialogResult.OK)
            {
                RForm.Dispose();

                return;
            }
            try
            {
                Object display = true;
                Outlook._Application olook = new Outlook.Application();
                Outlook._MailItem omail = olook.CreateItem(0);
                omail.HTMLBody = createMailBody(RForm.ReportCBX.Text);
                omail.Display(display);
            }
            finally
            {
                
                RForm.Dispose();
            }
        }
    }
   
}

File: VB.Net/Microsoft Office/PSSDK9/Form1.vb
Imports System
Imports System.IO




Public Class Form1
    Dim ps As PlanSwift9.PlanSwift
    Dim tlst As List(Of String)

    Sub LoadTakeOffItems(ByVal itm As PlanSwift9.IItem)
        Dim isItm As Boolean


        For idx = 0 To itm.ChildCount - 1
            Dim citm As PlanSwift9.IItem = itm.ChildItem(idx)
            If citm.GetProperty("type").ResultAsString = "Folder" Then
                LoadTakeOffItems(citm)
            Else
                isItm = citm.GetPropertyResultAsBoolean("IsItem", False)

                If (isItm = True) Then
                    tlst.Add(citm.GUID)
                End If
                If citm.ChildCount > 0 Then
                    LoadTakeOffItems(citm)
                End If
            End If
        Next

    End Sub
    Sub itemtoword(ByVal aitm As PlanSwift9.IItem, ByVal tbl As Microsoft.Office.Interop.Word.Table, ByVal ItmType As String, ByVal rowidx As Integer)
        If rowidx > 20 Then
            tbl.Rows.Add()
        End If
        If ItmType = "Digitizer" Then
            tbl.Cell(rowidx, 1).Range.Font.Bold = 1
            tbl.Cell(rowidx, 1).Range.Font.Italic = 0
        Else
            tbl.Cell(rowidx, 1).Range.Font.Bold = 0
            tbl.Cell(rowidx, 1).Range.Font.Italic = 1
        End If
        'Item Name
        tbl.Cell(rowidx, 1).Range.Text = aitm.Name
        'Item Qty
        tbl.Cell(rowidx, 2).Range.Text = aitm.GetProperty("Qty").ResultAsString
        'item Units
        tbl.Cell(rowidx, 3).Range.Text = aitm.GetProperty("Qty").Units
        'item Price Each
        tbl.Cell(rowidx, 4).Range.Text = aitm.GetPropertyResultAsString("Price Each", "")
        'item Price Total
        tbl.Cell(rowidx, 5).Range.Text = aitm.GetPropertyResultAsString("Price Total", "")

    End Sub
    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click
        Form2.ShowDialog()
        If Form2.DialogResult <> vbOK Then
            Exit Sub
        End If
        Dim Template As Object = Directory.GetCurrentDirectory() & "\Includes\SampleQuote.dotx"
        Dim newTemplate As Object = False
        Dim DocType As Object = 0
        Dim isVisible As Object = True
        Dim msword = New Microsoft.Office.Interop.Word.Application
        Dim msdoc As Microsoft.Office.Interop.Word.Document


        msword.Visible = False

        msdoc = msword.Documents.Add(Template, newTemplate, DocType, isVisible)
        Dim tbl As Microsoft.Office.Interop.Word.Table = msdoc.Tables.Item(2)

        Dim ReportType As String = Form2.ComboBox1.Text
        ps.BeginUpdate()

        Dim rowidx As Integer = 1
        If ReportType = "Digitizer Items Only" Then
            For idx = 0 To tlst.Count - 1
                Dim isDigitizer As Boolean = False
                Dim aitm As PlanSwift9.IItem = ps.GetItem(tlst.Item(idx))
                If aitm.GetPropertyResultAsBoolean("IsArea", False) Then
                    isDigitizer = True
                End If
                If aitm.GetPropertyResultAsBoolean("IsLinear", False) Then
                    isDigitizer = True
                End If
                If aitm.GetPropertyResultAsBoolean("IsSegment", False) Then
                    isDigitizer = True
                End If
                If aitm.GetPropertyResultAsBoolean("IsCount", False) Then
                    isDigitizer = True
                End If

                If isDigitizer Then
                    rowidx = rowidx + 1
                    itemtoword(aitm, tbl, "Digitizer", rowidx)
                End If
                ProgressBar1.Increment(idx)
            Next
        End If
        If ReportType = "Parts Only" Then
            For idx = 0 To tlst.Count - 1
                Dim aitm As PlanSwift9.IItem = ps.GetItem(tlst.Item(idx))
                If aitm.GetPropertyResultAsBoolean("IsPart", False) Then
                    rowidx = rowidx + 1
                    itemtoword(aitm, tbl, "Part", rowidx)
                End If
                ProgressBar1.Increment(idx)
            Next
        End If
        If ReportType = "Digitizer Items w/parts" Then
            
            For idx = 0 To tlst.Count - 1
                Dim aitm As PlanSwift9.IItem = ps.GetItem(tlst.Item(idx))
                rowidx = rowidx + 1
                If aitm.GetPropertyResultAsBoolean("isPart", False) Then
                    rowidx = rowidx + 1
                    itemtoword(aitm, tbl, "Part", rowidx)
                Else
                    rowidx = rowidx + 1
                    itemtoword(aitm, tbl, "Digitizer", rowidx)
                End If
                ProgressBar1.Increment(idx)
            Next
        End If
        ProgressBar1.Value = 0
        ps.EndUpdate()

        msword.Visible = True
        msdoc = Nothing
        msword = Nothing
    End Sub

    Private Sub Form1_FormClosed(ByVal sender As Object, ByVal e As System.Windows.Forms.FormClosedEventArgs) Handles Me.FormClosed
        ps = Nothing
    End Sub

    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        ps = New PlanSwift9.PlanSwift
        Dim tpath As PlanSwift9.IItem
        tpath = ps.GetItem(ps.Root.FullPath & "\Job\Takeoff")
        tlst = New List(Of String)
        'load all "Takeoff Items" into a list
        LoadTakeOffItems(tpath)
        ProgressBar1.Minimum = 0
        ProgressBar1.Maximum = tlst.Count
    End Sub
    Sub ItemToExcel(ByVal aitm As PlanSwift9.IItem, ByVal xlsheet As Microsoft.Office.Interop.Excel.Worksheet, ByVal rowidx As Integer)
        Dim row As Object = rowidx
        Dim xlcolumn As Object = 1
        If rowidx > 36 Then
            xlsheet.Cells.Item(row, xlcolumn).EntireRow.Insert()
        End If
        'QtyCell
        xlsheet.Cells.Item(rowidx, 1).value = aitm.GetPropertyResultAsString("Qty", "")
        'QtyUnits
        xlsheet.Cells.Item(rowidx, 2).value = aitm.GetProperty("Qty").Units
        'Item Number
        xlsheet.Cells.Item(rowidx, 3).value = aitm.GetPropertyResultAsString("Item #", "")
        'Item Name
        xlsheet.Cells.Item(rowidx, 4).value = aitm.Name
        'Item Price Each
        xlsheet.Cells.Item(rowidx, 6).value = aitm.GetPropertyResultAsString("Price Each", "")
        'Item Price Total
        xlsheet.Cells.Item(rowidx, 7).value = aitm.GetPropertyResultAsString("Price Total", "")
    End Sub
    Private Sub Button3_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button3.Click
        Form2.ShowDialog()
        If Form2.DialogResult <> vbOK Then
            Exit Sub
        End If
        Dim excel As New Microsoft.Office.Interop.Excel.Application
        Dim xlbook As Microsoft.Office.Interop.Excel.Workbook
        Dim Template As Object = Directory.GetCurrentDirectory & "\Includes\Estimate.xlt"
        Dim rowidx As Integer = 17
        xlbook = excel.Workbooks.Add(Template)
        Dim xlsheet As Microsoft.Office.Interop.Excel.Worksheet
        xlsheet = xlbook.Worksheets.Item(1)

        Dim ReportType = Form2.ComboBox1.Text

        If ReportType = "Digitizer Items Only" Then
            For idx = 0 To tlst.Count - 1
                Dim isDigitizer As Boolean = False
                Dim aitm As PlanSwift9.IItem = ps.GetItem(tlst.Item(idx))
                If aitm.GetPropertyResultAsBoolean("IsArea", False) Then
                    isDigitizer = True
                End If
                If aitm.GetPropertyResultAsBoolean("IsLinear", False) Then
                    isDigitizer = True
                End If
                If aitm.GetPropertyResultAsBoolean("IsSegment", False) Then
                    isDigitizer = True
                End If
                If aitm.GetPropertyResultAsBoolean("IsCount", False) Then
                    isDigitizer = True
                End If

                If isDigitizer Then
                    rowidx = rowidx + 1
                    ItemToExcel(aitm, xlsheet, rowidx)
                End If
                ProgressBar1.Increment(idx)
            Next
        End If
        If ReportType = "Parts Only" Then
            For idx = 0 To tlst.Count - 1
                Dim aitm As PlanSwift9.IItem = ps.GetItem(tlst.Item(idx))
                If aitm.GetPropertyResultAsBoolean("IsPart", False) Then
                    rowidx = rowidx + 1
                    ItemToExcel(aitm, xlsheet, rowidx)
                End If
                ProgressBar1.Increment(idx)
            Next
        End If
        If ReportType = "Digitizer Items w/parts" Then

            For idx = 0 To tlst.Count - 1
                Dim aitm As PlanSwift9.IItem = ps.GetItem(tlst.Item(idx))
                rowidx = rowidx + 1
                ItemToExcel(aitm, xlsheet, rowidx)
                ProgressBar1.Increment(idx)
            Next
        End If
        ProgressBar1.Value = 0
        excel.Visible = True
        excel = Nothing
        xlsheet = Nothing
        xlbook = Nothing
    End Sub
 
    Private Sub Button4_Click(ByVal sender As System.Object, ByVal e As System.EventArgs)
       
    End Sub
    Private Function createMailbody(ByVal RT As String) As String
        Dim body As String
        Dim itm As PlanSwift9.IItem
        body = "<body>" & vbCr
        body = body & "<h1>Report Type: " & RT & "</h1>" & vbCr
        body = body & "<table style=" & Chr(34) & "width:auto; font-weight:Bold; font-family:Tahoma; font-size:14px;" & Chr(34) & " > " & vbCr
        body = body & "<tr>" & vbCr
        body = body & "<td style=" & Chr(34) & "width:200px; font-weight:Bold; font-family:Tahoma; font-size:14px;" & Chr(34) & " > "
        body = body & "Name"
        body = body & "</td>" & vbCr
        body = body & "<td style=" & Chr(34) & "width:100px; font-weight:Bold; font-family:Tahoma; font-size:14px;" & Chr(34) & " > "
        body = body & "Qty"
        body = body & "</td>" & vbCr
        body = body & "<td style=" & Chr(34) & "width:100px; font-weight:Bold; font-family:Tahoma; font-size:14px;" & Chr(34) & " > "
        body = body & "Price Each"
        body = body & "</td>" & vbCr
        body = body & "<td style=" & Chr(34) & "width:100px; font-weight:Bold; font-family:Tahoma;font-size:14px;" & Chr(34) & " > "
        body = body & "Price Total"
        body = body & "</td>" & vbCr
        body = body & "</tr>" & vbCr
        If RT = "Digitizer Items Only" Then
            For idx = 0 To tlst.Count - 1
                itm = ps.GetItem(tlst(idx))
                If itm.GetPropertyResultAsBoolean("isPart", False) = False Then
                    body = body & "<tr><td style=" & Chr(34) & "font-size:12px; font-weight:normal;" & Chr(34) & ">" & itm.Name & "</td>"
                    body = body & "<td style=" & Chr(34) & "font-size:12px; font-weight:normal;" & Chr(34) & ">" & itm.GetPropertyResultAsString("Qty", "") & "</td>"
                    body = body & "<td style=" & Chr(34) & "font-size:12px; font-weight:normal;" & Chr(34) & ">" & itm.GetPropertyResultAsString("Price Each", "") & "</td>"
                    body = body & "<td style=" & Chr(34) & "font-size:12px; font-weight:normal;" & Chr(34) & ">" & itm.GetPropertyResultAsString("Price Total", "") & "</td>"
                    body = body & "</tr>"
                End If

            Next

        End If
        If RT = "Parts Only" Then
            For idx = 0 To tlst.Count - 1
                itm = ps.GetItem(tlst(idx))
                If itm.GetPropertyResultAsBoolean("isPart", False) = True Then
                    body = body & "<tr><td style=" & Chr(34) & "font-size:12px; font-weight:normal;" & Chr(34) & ">" & itm.Name & "</td>"
                    body = body & "<td style=" & Chr(34) & "font-size:12px; font-weight:normal;" & Chr(34) & ">" & itm.GetPropertyResultAsString("Qty", "") & "</td>"
                    body = body & "<td style=" & Chr(34) & "font-size:12px; font-weight:normal;" & Chr(34) & ">" & itm.GetPropertyResultAsString("Price Each", "") & "</td>"
                    body = body & "<td style=" & Chr(34) & "font-size:12px; font-weight:normal;" & Chr(34) & ">" & itm.GetPropertyResultAsString("Price Total", "") & "</td>"
                    body = body & "</tr>"
                End If

            Next

        End If
        If RT = "Digitizer Items w/parts" Then
            For idx = 0 To tlst.Count - 1
                itm = ps.GetItem(tlst(idx))
                If itm.GetPropertyResultAsBoolean("isPart", False) = False Then
                    body = body & "<tr><td style=" & Chr(34) & "font-size:12px; font-weight:normal; color:#007dc3;" & Chr(34) & ">" & itm.Name & "</td>"
                    body = body & "<td style=" & Chr(34) & "font-size:12px; font-weight:normal; color:#007dc3;" & Chr(34) & ">" & itm.GetPropertyResultAsString("Qty", "") & "</td>"
                    body = body & "<td style=" & Chr(34) & "font-size:12px; font-weight:normal; color:#007dc3;" & Chr(34) & ">" & itm.GetPropertyResultAsString("Price Each", "") & "</td>"
                    body = body & "<td style=" & Chr(34) & "font-size:12px; font-weight:normal; color:#007dc3;" & Chr(34) & ">" & itm.GetPropertyResultAsString("Price Total", "") & "</td>"
                    body = body & "</tr>"
                End If
                If itm.GetPropertyResultAsBoolean("isPart", False) = True Then
                    body = body & "<tr><td style=" & Chr(34) & "font-size:12px; font-weight:normal; color:#0000FF;" & Chr(34) & ">" & itm.Name & "</td>"
                    body = body & "<td style=" & Chr(34) & "font-size:12px; font-weight:normal; color:#0000FF;" & Chr(34) & ">" & itm.GetPropertyResultAsString("Qty", "") & "</td>"
                    body = body & "<td style=" & Chr(34) & "font-size:12px; font-weight:normal; color:#0000FF;" & Chr(34) & ">" & itm.GetPropertyResultAsString("Price Each", "") & "</td>"
                    body = body & "<td style=" & Chr(34) & "font-size:12px; font-weight:normal; color:#0000FF;" & Chr(34) & ">" & itm.GetPropertyResultAsString("Price Total", "") & "</td>"
                    body = body & "</tr>"
                End If

            Next
        End If


        createMailbody = body
    End Function
    Private Sub Button2_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button2.Click
        Form2.ShowDialog()
        If Form2.DialogResult <> vbOK Then
            Exit Sub
        End If
        Dim olook As Microsoft.Office.Interop.Outlook._Application
        Dim display As Object
        display = True
        olook = New Microsoft.Office.Interop.Outlook.Application
        Dim omail As Microsoft.Office.Interop.Outlook._MailItem
        omail = olook.CreateItem(0)
        omail.HTMLBody = createMailbody(Form2.ComboBox1.Text)
        omail.Display(display)
    End Sub
End Class
